### DISCUSSION TITLE
Dynamic answer file naming

### DISCUSSION THREAD

**Discussion by nilsdebruin** [copier-org/copier#860] [Ideas]
Hi all!

It is possible to define an answer file in the copier.yml like for example:

`_answers_file: .copier-answers-copier-template.yml`

It would be really nice if this could be extended with a dynamic value that you could set as a jinja variable like this:

`_answers_file: .copier-answers-copier-template-{{module_name}}.yml`

Where module_name is an answer you can give in copier.yml like:

```
module_name:
  type: str
```

This would make It possible to create multiple instances of a template in the same structure like:

```
root_level/{{module_name_1}}
root_level/{{module_name_2}}
```

This would create two answer files:

```
.copier-answers-copier-template-module_name_1.yml
.copier-answers-copier-template-module_name_2.yml
```

These template instances could then be updated based on these answer files.

**Comment by yajo** [comment]
Can't that be done already?

**Reply by nilsdebruin** [reply]
That would be great, but unfortunately when I implement this, the answer file will be named `.copier-answers-copier-template-{{module_name}}.yml` without the module_name which you supply. Am I missing something? Thanks for taking the time to answer and looking into this!

**Reply by yajo** [reply]
Then that's probably a bug because other options let you do that. I'll convert this to issue for you, so you can track it there (or fix it! 😜).

Thanks!

**Reply by nilsdebruin** [reply]
Great thanks @yajo!

**Reply by nilsdebruin** [reply]
This has been fixed in pull request: https://github.com/copier-org/copier/pull/1027 I have validated that this works as expected, @icj217 and @yajo thanks for fixing this!

**Comment by nilsdebruin** [comment]
Has been fixed
